test/integration/kubernetes_deploy_test.rb

```diff
@@ -27,6 +27,51 @@ def test_full_hello_cloud_set_deploy_succeeds
     assert_logs_match(%r{ConfigMap/hello-cloud-configmap-data\s+Available}, 1)
   end
 
+  def test_service_account_predeployed_before_unmanaged_pod
+    # Add an undefined service account in unmanaged pod
+    service_account_name = "unknown-service-account"
```

Comment:
Is the first part of this test adding much value?

The failure here comes from Kubernetes - we know the deployment will fail if it mentions a missing service account, so we're not testing k8s-deploy specific functionality. Unless I'm missing something, I'd suggest deleting this test case and keeping only the success case.

Reply:
I agree with @stefanmb on this.

Reply:
Agree. Removed the first part of the test which does not test feature provided kubernetes-deploy